src/middlewared/middlewared/plugins/apps/upgrade.py

```diff
@@ -10,16 +11,54 @@
 from .version_utils import get_latest_version_from_app_versions
 
 
+logger = logging.getLogger('app_lifecycle')
+
+
 class AppService(Service):
 
     class Config:
         namespace = 'app'
         cli_namespace = 'app'
 
+    @private
+    def take_snapshot_of_hostpath(self, app, snapshot_hostpath):
+        app_info = self.middleware.call_sync('app.get_instance', app) if isinstance(app, str) else app
+        host_path_mapping = self.middleware.call_sync('app.get_hostpaths_datasets', app_info['name'])
+        # Stop the app itself before we attempt to take snapshots
+        self.middleware.call_sync('app.stop', app_info['name']).wait_sync()
+        if not snapshot_hostpath:
```

Comment:
This `if not snapshot_hostpath` should be the first thing we do in this method, right? No reason to check all the other information if that's not provided since we return early.

Reply:
We still want to stop the app, i have updated the method name so it better reflects what is being done as it was not clear otherwise